README.md

```diff
@@ -1,108 +1,204 @@
+
+```yaml
+apiVersion: networking.scs.community/v1
+kind: Subnet
+metadata:
+  name: mysubnet
+spec:
+  # ...
+---
+apiVersion: kubernetes.scs.community/v1
+kind: Kubernetes
+metadata:
+  name: mykubernetes
+spec:
+  forProvider:
+    subnetRef:
+      name: mysubnet
+  # ...
+```
+
+This is obviously desirable from a user's perspective.
+However, unfortunately, it is also much more work than the SCS project can
+realistically build and maintain in the short/medium term.
+
+It also comes with the requirement to make many tough trade-off decisions.
+For example:
+
+Provider "A" offers to hide Kubernetes API server endpoints from the public
+internet, utilizing some sort of bastion host. Provider "B" instead implements
+IP based firewall blocking on the public endpoints. Provider "C" does neither.  
+Should the API follow either provider "A" or "B"? Should both approaches be
+implemented, but as optional features? If any of these approaches is defined
+to be a mandatory feature to support, provider "C" cannot be compliant.
+
+Any choice brings significant disadvantages:
+
+- If such features are included as features that are mandatory to support,
+  some providers may have difficulties adopting the API.
+- If such features are included as optional features, the ability to migrate
+  from one provider to another suffers significantly. Without this ability,
+  users also may opt to use provider-specific API's, instead.
+- If such features are excluded, the API becomes overall less useful for the
+  users who may opt to use more powerful provider-specific API's, instead.
+
+In sum: Going this route would be technically the best thing to do, yet does
+not seam feasible given tough trade-offs and limited resources.  
+If the opportunity arises to partner with some other organization with a lot
+of staff and resources, this option may be reevaluated, though.
+
+</details>
+
+## The chosen approach (for now)
+
+Goal: **Provide a "semantically" consistent API modelling most cloud resources
+that are in scope for SCS**.
+
+### Kubernetes API
+
+Instead of creating SCS-specific API idioms and building the implementation
+from scratch, the Kubernetes API will be "reused". Essentially, the Kubernetes
+API is just an opinionated REST API which has opinions on how a resource
+is defined, how it looks like, how it is reconciled/handled, how AuthN/AuthZ
+can be implemented. The Kubernetes ecosystem provides much tooling for working
+with such (custom) resource definitions: For creating the definitions
+themselves, building controllers, making them discoverable and deployable.
+
+As such, Kubernetes is a great choice for building any sort of resource
+management API - with some caveats regarding its deployment and the legacy
+of starting off as container orchestration tooling.
+
+### Crossplane tooling
+
+Crossplane even extends the Kubernetes API with "Composite Resource Definitions"
```

Comment:
it might be good to link to the definition here for readers that are not familiar with this crossplane implementation detail?

I believe this link should maybe do the job? https://docs.crossplane.io/latest/concepts/compositions/ (maybe better link to a specific version in case the "latest" docs might change in the future).

Reply:
Did so in 96833561574624eb58b63bbd27a72159829ab34e.